tests/pr_test.go

```diff
@@ -2,44 +2,55 @@
 package test
 
 import (
+	"log"
+	"os"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/cloudinfo"
+	"github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/common"
 	"github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/testhelper"
 )
 
-// Use existing resource group
-const resourceGroup = "geretain-test-resources"
-const completeExampleDir = "examples/complete"
+const yamlLocation = "../common-dev-assets/common-go-assets/common-permanent-resources.yaml"
+const resourceGroup = "geretain-test-observability-instances"
+const solutionInstanceTerraformDir = "solutions/instances"
+const region = "us-south"
 
-func setupOptions(t *testing.T, prefix string, dir string) *testhelper.TestOptions {
-	options := testhelper.TestOptionsDefaultWithVars(&testhelper.TestOptions{
-		Testing:       t,
-		TerraformDir:  dir,
-		Prefix:        prefix,
-		ResourceGroup: resourceGroup,
-	})
-	return options
-}
+var sharedInfoSvc *cloudinfo.CloudInfoService
+var permanentResources map[string]interface{}
 
-func TestRunCompleteExample(t *testing.T) {
-	t.Parallel()
+func TestMain(m *testing.M) {
+	sharedInfoSvc, _ = cloudinfo.NewCloudInfoServiceFromEnv("TF_VAR_ibmcloud_api_key", cloudinfo.CloudInfoServiceOptions{})
 
-	options := setupOptions(t, "mod-template", completeExampleDir)
+	// Read the YAML file contents
+	var err error
+	permanentResources, err = common.LoadMapFromYaml(yamlLocation)
+	if err != nil {
+		log.Fatal(err)
+	}
 
-	output, err := options.RunTestConsistency()
-	assert.Nil(t, err, "This should not have errored")
-	assert.NotNil(t, output, "Expected some output")
+	os.Exit(m.Run())
 }
 
-func TestRunUpgradeExample(t *testing.T) {
+func TestRunInstanceSolution(t *testing.T) {
 	t.Parallel()
 
-	options := setupOptions(t, "mod-template-upg", completeExampleDir)
+	options := testhelper.TestOptionsDefault(&testhelper.TestOptions{
+		Testing:       t,
+		TerraformDir:  solutionInstanceTerraformDir,
+		Region:        region,
+		ResourceGroup: resourceGroup,
+	})
 
-	output, err := options.RunTestUpgrade()
-	if !options.UpgradeTestSkipped {
-		assert.Nil(t, err, "This should not have errored")
-		assert.NotNil(t, output, "Expected some output")
+	options.TerraformVars = map[string]interface{}{
+		"cos_instance_access_tags": permanentResources["accessTags"],
+		"existing_kms_guid":        permanentResources["hpcs_south"],
+		"existing_resource_group":  true,
+		"resource_group_name":      options.ResourceGroup,
 	}
+
+	output, err := options.RunTestConsistency()
+	assert.Nil(t, err, "This should not have errored")
+	assert.NotNil(t, output, "Expected some output")
```

Comment:
You should use the schematics test to test the DA here, so it can use private endpoints. For example see [here](https://github.com/terraform-ibm-modules/terraform-ibm-scc-da/blob/935e3a0a6ef1da42306e2f3c1c39aa200d07b6fb/tests/pr_test.go#L43-L80).

Then for upgrade test you can use the standard upgrade test, but you will need to use pubilc endpoints. See [here](https://github.com/terraform-ibm-modules/terraform-ibm-scc-da/blob/935e3a0a6ef1da42306e2f3c1c39aa200d07b6fb/tests/pr_test.go#L82-L104).

Reply:
Done